src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -406,7 +414,7 @@ internal void PriorityValueChanged(AvaloniaProperty property, int priority, obje
             {
                 RaisePropertyChanged(property, oldValue, newValue, (BindingPriority)priority);
 
-                Logger.Verbose(
+                Logger.TryGetLogger(LogEventLevel.Verbose)?.Log(
```

Comment:
This syntax would look way better in other changed parts of the code. We should probably also remove the direct way of writing to log, so people won't accidentally cause unnecessary allocations.

Reply:
Problem is mostly caused by params array allocation.
I can use this pattern in one other place, but for a few other places I have to use boolean returning variant since we also do extra work if the logger is enabled (like fetching the description).


I think I might just add more generic parameters and expand to params in the logger itself. Maybe I should remove params overload completely? Otherwise somebody adding one extra parameter to the existing logging call can unknowingly introduce allocations.

Reply:
:+1: for removing `params`